modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/editor/BalloonEditor.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import {Editor} from './Editor';
+
+const BalloonEditor = ({config = {}, contents, name, ...otherProps}) => {
+	const defaultExtraPlugins = 'balloontoolbar,floatingspace';
+
+	const getConfig = () => {
+		const extraPlugins = config.extraPlugins
+			? `${config.extraPlugins}`
+			: '';
+
+		return {
+			...config,
+			extraAllowedContent: '*',
+			extraPlugins: `${extraPlugins}${defaultExtraPlugins}`,
+		};
+	};
+
+	return (
+		<Editor
```

Comment:
Just a thought... couldn't we use `ClassicEditor` as a base here? It feels like we've added a lot of fixes in there that we might need to replicate here every time and that could be avoided if we already use a better base.

Just wondering why the decision of building this from the ground up and not on top of it.

Reply:
I mainly avoided it, because `BalloonEditor` won't support as many features as `ClassicEditor`, for example [toolbarSets](https://github.com/liferay/liferay-portal/blob/9c07f7bca2bbde25a8cbcf2c84b6b311fb60e0f7/modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/editor/ClassicEditor.js#L38) which probably won't be defined the same way (considering that "BalloonToolbars" aren't exactly the same as regular toolbars in CKEditor), I wasn't sure if we wanted to maintain [this global API](https://github.com/liferay/liferay-portal/blob/9c07f7bca2bbde25a8cbcf2c84b6b311fb60e0f7/modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/editor/ClassicEditor.js#L101-L108) either and I wanted to avoid [this resize handler](https://github.com/liferay/liferay-portal/blob/9c07f7bca2bbde25a8cbcf2c84b6b311fb60e0f7/modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/editor/ClassicEditor.js#L101-L108)  - but I'll try building `BalloonEditor` on top of `ClassicEditor` and see how it goes.

Reply:
No, that's okay, sounds good to me. Just wanted to make sure we had a good reason for this. We can always create a common base if we see a lot of duplication between the editors.

Let's just add the release-ignore markers and push this forward if ready!

Reply:
:+1: